exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -145,132 +170,234 @@ func (e *Exchange) SetDefaults() {
 
 // Setup sets user exchange configuration settings
 func (e *Exchange) Setup(exch *config.Exchange) error {
-	err := exch.Validate()
-	if err != nil {
+	if err := exch.Validate(); err != nil {
 		return err
 	}
 	if !exch.Enabled {
 		e.SetEnabled(false)
 		return nil
 	}
-	err = e.SetupDefaults(exch)
-	if err != nil {
+	if err := e.SetupDefaults(exch); err != nil {
 		return err
 	}
 
-	wsRunningURL, err := e.API.Endpoints.GetURL(exchange.WebsocketSpot)
-	if err != nil {
-		return err
+	// Migrate legacy URLs to the new format
+	em := e.API.Endpoints.GetURLMap()
+	if em[exchange.RestSpot.String()] == "https://poloniex.com" {
+		if err := e.API.Endpoints.SetRunningURL(exchange.RestSpot.String(), apiURL); err != nil {
+			return err
+		}
+		exch.API.Endpoints[exchange.RestSpot.String()] = apiURL
+	}
+	if em[exchange.WebsocketSpot.String()] == "wss://api2.poloniex.com" {
+		if err := e.API.Endpoints.SetRunningURL(exchange.WebsocketSpot.String(), websocketURL); err != nil {
+			return err
+		}
+		exch.API.Endpoints[exchange.WebsocketSpot.String()] = websocketURL
 	}
```

Comment:
🚧 Change request:
Pretty sure I saw this requested before.
Please move to a config version
```suggestion
```

Reply:
🚧 Change request:
@samuael I don't understand your reason for rejecting.
I understand what the code is doing.
Please move to a config version.

Reply:
I meant that this should be a single time upgrade to the config using a config version upgrade like v12.

What you're saying with this code is that the user has a config.json with old URLs in it which are no longer correct.
And when we see those, we want to change them to the new URLs.
But if they have any other configuration, we want to leave it alone.
Is that correct?

That's exactly what [v8 did for bitmex](https://github.com/thrasher-corp/gocryptotrader/blob/master/config/versions/v8/v8.go)

So you do exactly the same thing for this.

Reply:
@samuael You haven't replied to this thread.